### ISSUE TITLE
[web] Barcode scanning not working

### ISSUE THREAD

**Issue by codewithbayanda** [juliansteenbakker/mobile_scanner#835]
I event tried the examples provided by the package but its just stays about 3 seconds then it went black[camera frame not working]

and also this warning [The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead]

**Comment by navaronbracke** [comment]
What Flutter version are you using? The `dart:ui_web` warning was fixed on stable a while ago.

Are you using `mobile_scanner` version `3.5.1`? Are you adding the required script to your index.html? (see https://github.com/juliansteenbakker/mobile_scanner#web)

Is this on a specific browser?

**Comment by ettaegbe** [comment]
Facing the same issue.

`The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead. `

This causes the camera to not work during production.

**Comment by navaronbracke** [comment]
I updated the package to the new `ui_web.platformViewRegistry` in https://github.com/juliansteenbakker/mobile_scanner/pull/916

Hopefully, we can get that landed soon.
I can also prepare a smaller change that fixes _only_ the deprecation message, that will help too in this case.

**Comment by delizondo** [comment]
I'm using `mobile_scanner: ^4.0.0`, and I've already injected the script 
```<script src="https://unpkg.com/@zxing/library@0.19.1" type="application/javascript"></script>``` in the `index.html` file. However, when I try to open the scanned in a web browser, it just failed to load, no error, no nothing, it just never opens the camera. 

Any ideas?

![Screenshot 2024-01-31 at 12 13 21 PM](https://github.com/juliansteenbakker/mobile_scanner/assets/1301103/2356fa38-8bf9-4e39-897e-40df9db29926)

**Comment by navaronbracke** [comment]
@delizondo Did you put the script either in the `<head>` or at the top of the `<body>` (before the Flutter Web bootstrapping code that loads the application) ?

You do not see any errors in the browser console at all?

**Comment by delizondo** [comment]
@navaronbracke yes, it's part of the `<head>`, and no errors are displayed in the console 
![Screenshot 2024-02-02 at 8 34 11 AM](https://github.com/juliansteenbakker/mobile_scanner/assets/1301103/efee488f-08db-4438-a773-ed7751b69a0a)
![Screenshot 2024-02-02 at 8 37 09 AM](https://github.com/juliansteenbakker/mobile_scanner/assets/1301103/e3253aec-dbc8-4a51-a6b6-edbd3bc60b1d)

**Comment by navaronbracke** [comment]
From the above I can see that you use:
- version 0.20.0 of ZXing (the example app uses version 0.19.1)
- you use the minified version? I don't think the version from the example app is stripped?
- you also have a warning about third party cookies, but I am not sure if that is related

**Comment by delizondo** [comment]
@navaronbracke 
I've changed it to `<script src="https://unpkg.com/@zxing/library@0.19.1" type="application/javascript"></script>
`, and also tested with `<script src="https://unpkg.com/@zxing/library@0.19.1/umd/index.min.js" type="application/javascript"></script>` but the problem still persist. The only difference is that now the Cookies message no longer appear. 

If I try to run the same example in iOS or Android, everything works fine, is just when I try to run it in Web in my laptop

Not sure if it helps, but I'm running Chrome Version 121.0.6167.139 (Official Build) (arm64) in a Mac M1. Also, attached is the flutter doctor, if it helps

```david.elizondo@David-Elizondo ~ % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.9, on macOS 14.3 23D56 darwin-arm64, locale
    en-CR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.86.0)
[✓] Connected device (4 available)
[✓] Network resources```

**Comment by navaronbracke** [comment]
Does the `mobile_scanner` example app also have this problem when you run it on your browser?

I do not think that the browser version would be the problem.
Sadly I cannot test the app on an M1, as I do not have access to such a device yet.

**Comment by delizondo** [comment]
Downloaded the example, and finally saw what I was doing wrong, it's working fine in web now :)

**Comment by navaronbracke** [comment]
Great to hear! Just for clarity, could you explain to me what the problem was? Then I might be able to offer better guidance to people who end up in the same situation as you did.

**Comment by Fe1ix97** [comment]
Hello, 
in my case the camera opens but does not scan any barcodes. I also tried with the basic example you provide on pubspec, but it doesn't work. On mobile everything ok instead

**Comment by navaronbracke** [comment]
With the new version 5.0.0-beta.3 there were some improvements for web. You could try with that version.

**Comment by juliansteenbakker** [comment]
Please let me know if it's still not working in the latest version.

**Comment by charlieaboumoussa** [comment]
Hello.
On mobile, scanning is very smooth (auto focus, accuracy, etc..) and super fast.

However on web, it is still very hard to scan the barcode. 
Camera doesn't focus well. It behaves differently on different devices and browsers (some browsers/devices auto focus, some don't)

Any idea if there will be a fix for this please?
Are we missing out on something? Is there some configuration that needs to be done for web?
Thanks

**Comment by angelru** [comment]
> Hello. On mobile, scanning is very smooth (auto focus, accuracy, etc..) and super fast.
> 
> However on web, it is still very hard to scan the barcode. Camera doesn't focus well. It behaves differently on different devices and browsers (some browsers/devices auto focus, some don't)
> 
> Any idea if there will be a fix for this please? Are we missing out on something? Is there some configuration that needs to be done for web? Thanks

Did you find a solution?
